autonomi/src/client/data/public.rs

```diff
@@ -44,100 +49,72 @@ impl Client {
         data: Bytes,
         payment_option: PaymentOption,
     ) -> Result<DataAddr, PutError> {
-        let now = ant_networking::target_arch::Instant::now();
-        let (data_map_chunk, chunks) = encrypt(data)?;
-        let data_map_addr = data_map_chunk.address();
-        debug!("Encryption took: {:.2?}", now.elapsed());
-        info!("Uploading datamap chunk to the network at: {data_map_addr:?}");
-
-        let map_xor_name = *data_map_chunk.address().xorname();
-        let mut xor_names = vec![map_xor_name];
-
-        for chunk in &chunks {
-            xor_names.push(*chunk.name());
-        }
-
-        // Pay for all chunks + data map chunk
-        info!("Paying for {} addresses", xor_names.len());
-        let receipt = self
-            .pay_for_content_addrs(xor_names.into_iter(), payment_option)
-            .await
-            .inspect_err(|err| error!("Error paying for data: {err:?}"))?;
-
-        // Upload all the chunks in parallel including the data map chunk
-        debug!("Uploading {} chunks", chunks.len());
-
-        let mut failed_uploads = self
-            .upload_chunks_with_retries(
-                chunks
-                    .iter()
-                    .chain(std::iter::once(&data_map_chunk))
-                    .collect(),
-                &receipt,
-            )
-            .await;
-
-        // Return the last chunk upload error
-        if let Some(last_chunk_fail) = failed_uploads.pop() {
-            tracing::error!(
-                "Error uploading chunk ({:?}): {:?}",
-                last_chunk_fail.0.address(),
-                last_chunk_fail.1
-            );
-            return Err(last_chunk_fail.1);
-        }
-
-        let record_count = chunks.len() + 1;
-
-        // Reporting
-        if let Some(channel) = self.client_event_sender.as_ref() {
-            let tokens_spent = receipt
-                .values()
-                .map(|(_proof, price)| price.as_atto())
-                .sum::<Amount>();
-
-            let summary = UploadSummary {
-                record_count,
-                tokens_spent,
-            };
-            if let Err(err) = channel.send(ClientEvent::UploadComplete(summary)).await {
-                error!("Failed to send client event: {err:?}");
+        match &self.mode {
+            ClientMode::ReadWrite(_) => {
+                let now = ant_networking::target_arch::Instant::now();
```

Comment:
Why introduce a mode here?

Reply:
The idea is we have a read only client (no wallet etc.) and a full client that can put so needs a wallet